### PR TITLE
Wrap Gas-Fired Absorption Heat Pump (GAHP)

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

Companion PR:
* https://github.com/NREL/OpenStudio/pull/4806

Link to relevant GitHub Issue(s) if appropriate:

Link to the **Linux.deb** installer to use for CI Testing. If not set, it will default to latest official release.
[OpenStudio Installer]: http://openstudio-ci-builds.s3-website-us-west-2.amazonaws.com/PR-4806/OpenStudio-3.6.0-alpha%2Be2671346b9-Ubuntu-20.04-x86_64.deb

This Pull Request is concerning:

 - [x] **Case 1 - `NewTest`:** a new test for a new model API class,

----------------------------------------------------------------------------------------------------------

### Case 1: New test for a new model API class

Please include which class(es) you are adding a test to specifically test for.
Include a link to the OpenStudio Pull Request in which you are adding the new classes, or the class itself if already on develop.

#### Work Checklist

The following has been checked to ensure compliance with the guidelines:

 - [ ] Tests pass either:
     - [ ] with official OpenStudio release (include version):
         - [ ] A matching OSM test has been added from the successful run of the Ruby one with the official OpenStudio release
         - [ ] The label `AddedOSM` has been added to this PR
         - [ ] All new `out.osw` have been committed

     - [ ] with current develop (incude SHA):
         - [ ] The label `PendingOSM` has been added to this PR
         - [ ] A matching OSM test has not yet been added because the official release is pending, but `model_tests.rb` has a TODO.
            ```ruby
            def test_airterminal_cooledbeam_rb
              result = sim_test('airterminal_cooledbeam.rb')
            end

            # TODO: To be added in the next official release after: 2.5.0
            # def test_airterminal_fourpipebeam_osm
            #   result = sim_test('airterminal_fourpipebeam.osm')
            # end
            ```
        - [ ] No `out.osw` have been committed as they need to be run with an official OpenStudio version

 - [ ] **Ruby test is stable**: when run multiple times on the same machine, it produces the same total site kBTU.
    Please paste the heatmap png generated after running the following commands:
     - [ ] I ensured that I assign systems/loads/etc in a repeatable manner (eg: if I assign stuff to thermalZones, I do `model.getThermalZones.sort_by{|z| z.name.to_s}.each do ...` so I am sure I put the same ZoneHVAC systems to the same zones regardless of their order)
     - [ ] I tested stability using `process_results.py` (see `python process_results.py --help` for usage).
    Please paste the text output or heatmap png generated after running the following commands:
        ```bash
        # Clean up all custom-tagged OSWs
        python process_results.py test-stability clean
        # Run your test 5 times in a row. Replace `testname_rb` (eg `airterminal_fourpipebeam_rb`)
        python process_results.py test-stability run -n testname_rb
        # Check that they all passed
        python process_results.py test-status --tagged
        # Check site kBTU differences
        python process_results.py heatmap --tagged

        ```

 - [ ] **Object has been added to `autosize_hvac.rb` to ensure the autosizedXXX values methods do work**

----------------------------------------------------------------------------------------------------------

### Review Checklist

 - [ ] Code style (indentation, variable names, strip trailing spaces)
 - [ ] Functional code review (it has to work!)
 - [ ] Matching OSM test has been added or `# TODO` added to `model_tests.rb`
 - [ ] Appropriate `out.osw` have been committed
 - [ ] Test is stable
 - [ ] Object is tested in `autosize_hvac` as appropriate
 - [ ] The appropriate labels have been added to this PR:
   - [ ] One of: `NewTest`, `TestFix`, `NewTestForExisting`, `Other`
   - [ ] If `NewTest`: add `PendingOSM` or `AddedOSM`

### CODE REVIEW

File: model/simulationtests/autosize_hvac.rb [hunk 1]
```diff
@@ -329,6 +329,13 @@ def curve_quadratic(model, c_1constant, c_2x, c_3xPOW2, minx, maxx, miny, maxy)
 # The Source Side Volume Flow Rate is reported only for WaterSource apparently
 cw_loop.addDemandBranchForComponent(plhp_clg)
 
+ffhp_airsource_clg = OpenStudio::Model::HeatPumpAirToWaterFuelFiredCooling.new(model)
+ffhp_airsource_clg.autosizeNominalCoolingCapacity
+ffhp_airsource_clg.autosizeDesignFlowRate
+ffhp_airsource_clg.autosizeDesignTemperatureLift
+ffhp_airsource_clg.setNominalAuxiliaryElectricPower(500)
+chw_loop.addSupplyBranchForComponent(ffhp_airsource_clg)
```
Comment (jmarrec) [inline model/simulationtests/autosize_hvac.rb:337]: Nice!

File: model/simulationtests/autosize_hvac.rb [hunk 2]
```diff
@@ -1047,6 +1064,16 @@ def curve_quadratic(model, c_1constant, c_2x, c_3xPOW2, minx, maxx, miny, maxy)
...
+  when 41
+    plhp_airsource_htg = OpenStudio::Model::HeatPumpAirToWaterFuelFiredHeating.new(model)
+    plhp_airsource_htg.autosizeNominalHeatingCapacity
+    plhp_airsource_htg.autosizeDesignFlowRate
+    plhp_airsource_htg.autosizeDesignTemperatureLift
+
+    plhp_airsource_clg = OpenStudio::Model::HeatPumpAirToWaterFuelFiredCooling.new(model)
+    plhp_airsource_clg.autosizeNominalCoolingCapacity
+    plhp_airsource_clg.autosizeDesignFlowRate
+    plhp_airsource_clg.autosizeDesignTemperatureLift
```
Comment (jmarrec) [inline model/simulationtests/autosize_hvac.rb]: Why is this here?

File: model_tests.rb [hunk 3]
```diff
@@ -586,6 +586,15 @@ def test_heatexchanger_desiccant_balancedflow_osm
...
   end
 
+  def test_heatpump_airtowater_fuelfired_rb
+    result = sim_test('heatpump_airtowater_fuelfired.rb')
+  end
+
+  # TODO: To be added in the next official release after: 3.5.1
+  # def test_heatpump_airtowater_fuelfired_osm
+  # result = sim_test('heatpump_airtowater_fuelfired.osm')
+  # end
```
Comment (jmarrec) [inline model_tests.rb:605]: good

File: model/simulationtests/autosize_hvac.rb (hunk 2 above)
Comment (jmarrec) [inline model/simulationtests/autosize_hvac.rb]:
```suggestion
```

**Review by jmarrec — CHANGES_REQUESTED** [review]

File: model/simulationtests/autosize_hvac.rb (hunk 2 above)
Comment (joseph-robertson) [reply model/simulationtests/autosize_hvac.rb]: We don't need to call out the autosize methods in the autosize_hvac test?

Comment (jmarrec) [reply model/simulationtests/autosize_hvac.rb]: That wasn't the reason I deleted this block. Above you have almost the same, except you then add them to a PlantLoop (as they should).

This block here achieves nothing but to create two extras orphaned objects. This is meant to add a ZoneHVAC component on a zone (the when 41 means the 41st zone)

Comment (joseph-robertson) [reply model/simulationtests/autosize_hvac.rb]: Ah. I must have stubbed out (and didn't finish) adding a ZoneHVAC component (like in heatpump_airtowater_fuelfired.rb).
